src/openforms/js/components/admin/form_design/registrations/zgw/ZGWOptionsFormFields.js

```diff
@@ -25,7 +26,21 @@ import {
 } from './fields';
 import {filterErrors} from './utils';
 
-const ZGWFormFields = ({name, apiGroupChoices, confidentialityLevelChoices}) => {
+/**
+ * Callback to invoke when the API group changes - used to reset the dependent fields.
+ */
+const onApiGroupChange = prevValues => ({
+  ...prevValues,
+  objecttype: '',
+  objecttypeVersion: undefined,
+});
```

Comment:
I'm not sure if this is actually needed, since this form still uses the old Objecttype and ObjecttypeVersion fields (simple inputs instead of select)

Reply:
this will get refactored as part of #4606, I'm gradually making my way down the various fields :D